File: Codeforces/1527/B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n;
    string s;
    cin >> t;
    while (t--) {
        cin >> n >> s;
        int zeros = count(s.begin(), s.end(), '0'), flips = 0;
        for (int i = 0; i < n / 2; ++i) {
            if (s[i] != s[n - i - 1]) ++flips;
        }
        if (flips == 0) {
            if (zeros % 2 == 0 || zeros == 1) {
                cout << "BOB\n";
            } else {
                cout << "ALICE\n";
            }
        } else if (flips == 1) {
            if (zeros - 1 == 1) {
                cout << "DRAW\n";
            } else {
                cout << "ALICE\n";
            }
        } else {
            cout << "ALICE\n";
        }
    }
    return 0;
}

File: Codeforces/1527/B.kt
private fun readLn() = readLine()!!
private fun readInt() = readLn().toInt()

fun main() {
    val T = readInt()
    for (t in 1..T) {
        val N = readInt()
        val S = readLn()
        val zeros = S.count { it == '0' }
        var flips = 0
        for (i in 0 until N / 2) {
            if (S[i] != S[N - i - 1]) ++flips
        }
        when (flips) {
            0 -> println(if (zeros % 2 == 0 || zeros == 1) "BOB\n" else "ALICE\n")
            1 -> println(if (zeros == 2) "DRAW\n" else "ALICE\n")
            else -> println("ALICE\n")
        }
    }
}